client/__tests__/e2e/e2eAnnotations.test.js

```diff
@@ -48,6 +84,87 @@ async function setup(config) {
   await waitByClass("autosave-complete");
 }
 
+describe("geneSET crud operations and interactions", () => {
+  test("genesets load from csv", async () => {
+    await goToPage(appUrlBase);
+
+    await assertGenesetExists(preExistingGenesetName);
+  });
+  test("brush on geneset mean", async () => {
+    await goToPage(appUrlBase);
+
+    await expandGeneset(meanExpressionBrushGenesetName);
+
+    const histBrushableAreaId = `histogram-${meanExpressionBrushGenesetName}-plot-brushable-area`;
+
+    const coords = await calcDragCoordinates(histBrushableAreaId, {
+      x1: 0.25,
+      y1: 0.5,
+      x2: 0.55,
+      y2: 0.5,
+    });
+
+    await drag(histBrushableAreaId, coords.start, coords.end);
+
+    const cellCount = await getCellSetCount(1);
+
+    expect(cellCount).toBe(meanExpressionBrushCellsSelected);
+  });
+  test("create a new geneset", async () => {
+    await goToPage(appUrlBase);
+
+    const genesetName = `test-geneset-foo-123`;
+    await assertGenesetDoesNotExist(genesetName);
+    await createGeneset(genesetName);
+    /* note: as of June 2021, the aria label is in the truncate component which clones the element */
+    await assertGenesetExists(genesetName);
+  });
+  test("edit geneset name", async () => {
+    await goToPage(appUrlBase);
+
+    await editGenesetName(editableGenesetName, editText);
+    await assertGenesetExists(newGenesetName);
+  });
+  test("delete a geneset", async () => {
+    await goToPage(appUrlBase);
+
+    await deleteGeneset(genesetToDeleteName);
+  });
+});
+
+describe("GENE crud operations and interactions", () => {
```

Comment:
I like that the tests focus on the happy path primarily.  But there are some very important happy path tests that you need to add, as they are historically where we see bugs.  I suggest:
* color-by gene set & gene in a gene set
* test on subset, not just whole (there is already a framework within this test file to do so)
* test with clip
* undo/redo when you create, edit or delete a geneset and/or a gene in a geneset